sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -170,8 +170,10 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog with QueryErrorsB
         u.tableNotFound(u.multipartIdentifier)
 
       case u: UnresolvedFunc =>
-        throw QueryCompilationErrors.noSuchFunctionError(
-          u.multipartIdentifier, u, u.possibleQualifiedName)
+        throw QueryCompilationErrors.unresolvedRoutineError(
+          u.multipartIdentifier,
+          Seq("system.builtin", "system.session"),
```

Comment:
why not put current database?

Reply:
Just current database? Could you show concrete example, please. Especially here (where should I take the current database here?).

Reply:
This code path is for reporting error of `DESCRIBE FUNCTION ...` etc., I think it does support persistent functions and the search path should include current database.